### PR TITLE
MAGECLOUD-2882: Duplication beetwen SCD_EXCLUDE_THEME and SCD_MATRIX

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
Added deprecation for the SCD_EXCLUDE_THEME variable. This variable became redundant after adding SCD_MATRIX.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-2882

### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGETWO-97873

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Config/Validator/Build/StageConfigDeprecatedVariables.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                    $error .= sprintf(' Use %s instead.', $deprecatedSchema[$key][Schema::SCHEMA_REPLACEMENT]);
+                }
+
+                $errors[$key] = $error;
+            }
+        }
+
+        if ($errors) {
+            return $this->resultFactory->create(Validator\ResultInterface::ERROR, [
+                'error' => 'Some configuration in your .magento.env.yaml file is deprecated.',
```
Comment (shiftedreality) [inline src/Config/Validator/Build/StageConfigDeprecatedVariables.php]: Shouldn't this be in plural?

File: src/Config/StageConfigInterface.php [hunk 2]
```diff
@@ -29,14 +29,18 @@ interface StageConfigInterface
...
-    const VAR_SCD_EXCLUDE_THEMES = 'SCD_EXCLUDE_THEMES';
     const VAR_SKIP_SCD = 'SKIP_SCD';
     const VAR_VERBOSE_COMMANDS = 'VERBOSE_COMMANDS';
     const VAR_SCD_ON_DEMAND = 'SCD_ON_DEMAND';
     const VAR_SKIP_HTML_MINIFICATION = 'SKIP_HTML_MINIFICATION';
     const VAR_SCD_MATRIX = 'SCD_MATRIX';
     const VAR_X_FRAME_CONFIGURATION = 'X_FRAME_CONFIGURATION';
 
+    /**
+     * @deprecated use SCD_MATRIX instead
```
Comment (shiftedreality) [inline src/Config/StageConfigInterface.php]: Add dot in the end

**Review by shiftedreality — APPROVED** [review]

File: src/Config/Validator/Build/StageConfigDeprecatedVariables.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/Config/Validator/Build/StageConfigDeprecatedVariables.php]: changed

File: src/Config/StageConfigInterface.php (hunk 2 above)
Comment (oshmyheliuk) [reply src/Config/StageConfigInterface.php]: added

**Review by BaDos — CHANGES_REQUESTED** [review]
Please add information about deprecated variable into the `.magento.env.yaml.dist` file

**Comment by oshmyheliuk** [comment]
@BaDos Information added

**Review by BaDos — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**